packages/fuel-indexer-graphql/src/dynamic.rs

```diff
@@ -0,0 +1,457 @@
+
+        let sort_object = sort_input_vals.into_iter().fold(
+            InputObject::new(format!("{entity_type}Sort")),
+            |input_obj, input_val| input_obj.field(input_val),
+        );
+
+        // For some reason, async-graphql does not implement the Hash
+        // trait on any of the type that we need for dynamic schemas.
+        // So we are essentially making a hash table ourselves for
+        // the filter and sort objects.
+        filter_object_list.push(filter_object);
+        filter_tracker.insert(entity_type.to_string(), filter_object_list.len() - 1);
+
+        sort_object_list.push(sort_object);
+        sorter_tracker.insert(entity_type.to_string(), sort_object_list.len() - 1);
+
+        // Additionally, because we cannot refer to the object fields directly and
+        // associate the field arguments to them, we iterate through the fields a
+        // second time and construct the fields for the dynamic schema and add the
+        // field arguments as well.
+        let mut fields = Vec::new();
+        for (field_name, field_type) in field_map {
+            if IGNORED_ENTITY_FIELD_TYPES.contains(&field_name.as_str()) {
+                continue;
+            }
+
+            if let Some(field_def) = Type::new(&field_type) {
+                let base_field_type = &field_def.base;
+                let nullable = field_def.nullable;
+
+                let field_type = match base_field_type {
+                    BaseType::Named(type_name) => {
+                        if nullable {
+                            TypeRef::named(type_name.to_string())
+                        } else {
+                            TypeRef::named_nn(type_name.to_string())
+                        }
+                    }
+                    BaseType::List(list_type) => {
+                        let inner_base_type = list_type.base.to_string();
+                        let nullable_inner = list_type.nullable;
+
+                        if nullable && nullable_inner {
+                            TypeRef::named_list(inner_base_type)
+                        } else if nullable && !nullable_inner {
+                            TypeRef::named_nn_list(inner_base_type)
+                        } else if !nullable && nullable_inner {
+                            TypeRef::named_list_nn(inner_base_type)
+                        } else {
+                            TypeRef::named_nn_list_nn(inner_base_type)
+                        }
+                    }
+                };
+
+                // Because the dynamic schema is set to only resolve introspection
+                // queries, we set the resolvers to return a dummy value.
+                let mut field = Field::new(
+                    field_name.clone(),
+                    field_type.clone(),
+                    move |_ctx: ResolverContext| {
+                        return FieldFuture::new(async move {
+                            Ok(Some(FieldValue::value(1)))
```

Comment:
There's a comment on lines 289-290 that describes the purpose; I can move it directly above, if desired.

```
 // Because the dynamic schema is set to only resolve introspection
 // queries, we set the resolvers to return a dummy value.
```

Reply:
Additional info:

Each field **requires** a resolver; believe me, I spent way too much time trying to get around this requirement. And depending on whether a user's indexer schema is nullable, we're unable to use `FieldValue::NONE` because the constraints of the `Field::new()` method require that the `FieldFuture` of any non-nullable field return a concrete type. So to get around all of that, we just return an arbitrary value because at the end of it all, these field resolver functions will never execute because the dynamic schema is just used for introspection.